src/parquet/arrow/writer.cc

```diff
@@ -342,29 +358,297 @@ Status FileWriter::Impl::WriteFlatColumnChunk(
   return Status::OK();
 }
 
+template <typename ParquetType, typename ArrowType>
+Status FileWriter::Impl::WriteListBatch(ColumnWriter* column_writer,
+    const ListArray* data, int64_t offset, int64_t length, int64_t num_levels,
```

Comment:
Why pass in `ListArray` here at all, and these functions below? All you need from the ListArray is the begin and end offset. 

Having list-related code specialized for individual types in general would be best avoided -- you construct the repetition and definition levels representing a certain number of logical "values" (per the discussion above), then write the correct number of values from the innermost leaf node. The trouble with this code is that it will need to be immediately refactored as soon as we are dealing in structs or more levels of lists. 

I suspect that TypedWriteBatch can be generalized or split up until submethods